source/extensions/transport_sockets/tls/io_handle_bio.cc

```diff
@@ -123,13 +126,18 @@ long io_handle_ctrl(BIO* b, int cmd, long num, void*) {
 BIO* BIO_new_io_handle(Envoy::Network::IoHandle* io_handle) {
   BIO* b;
   BIO_METHOD* meth = BIO_meth_new(BIO_get_new_index(), "io_handle");
+  bool ok = meth != nullptr;
+
```

Comment:
does it make sense to fail early if `meth == nullptr`?

Reply:
Well I tried to simplify control flow and maximize readability while addressing all the possible ways this code may fail in this code, by conditioning everything on the status we track in `ok`. Otherwise we'd get lots of early returns and if statements.. so I am leaning to keep this as is?

Reply:
Yeah, makes sense. I was just wondering if it's worth to execute the whole function if `meth == nullptr`. It looks like they don't check for null args, which may lead to a crash: https://github.com/openssl/openssl/blob/master/crypto/bio/bio_meth.c#L89

Reply:
resolving (John and me discussed offline)